### PR TITLE
Improve the Keda module testing strategy doc

### LINKED ISSUE #370: Update test strategy (CI/CD) documentation
**Description**
Keep test strategy [doc](https://github.com/kyma-project/keda-manager/blob/main/docs/contributor/04-10-ci-cd.md) up to date.

**Reason**
Test strategy should be documented and up to date to help contributors understand which tests are executed during development cycle.

### PR DESCRIPTION
**Pull request by grego952**
**Description**

Changes proposed in this pull request:

- Improve the testing strategy doc

**Related issue(s)**
Fixes #370

### CODE REVIEW

File: docs/contributor/04-10-ci-cd.md [hunk 1]
```diff
@@ -6,19 +6,18 @@ The following CI jobs are part of the development cycle. They verify the functio
...
-|[`pre-main-keda-manager-verify`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L175)|true|Installs Keda Manager, not using Lifecycle Manager, and applies the sample Keda CR in a k3d cluster. Executes smoke integration test of KEDA.  |
-|[`pre-keda-manager-operator-lint`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L61)|false|Is responsible for linting, static code analysis.|
+|[`PR Markdown Link Check / markdown-link-check (pull_request)`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/pr-markdown-link-check.yaml)|true|Checks if there are no broken links in the pull request `.md` files.|
+|[`pre-keda-manager-operator-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/keda-manager.yaml#L43)|true|Builds Keda operator's image and pushes it to the `dev` registry.|
+|[`pre-keda-manager-presubmit-scanner`](https://github.tools.sap/kyma/security-scanners#readme)|true|Checks if the repository doesn't contain any vulnerabilities.|
```
Comment (pPrecel) [inline docs/contributor/04-10-ci-cd.md:12]: I would suggest to adding missing tests described [here](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L18-L56). I think three of them are missing:

```
Keda verify / lint
Keda verify / unit-test
Keda verify / integration-test
```

**Review by pPrecel — APPROVED** [review]

File: docs/contributor/04-10-ci-cd.md [hunk 2]
```diff
@@ -6,19 +6,21 @@ The following CI jobs are part of the development cycle. They verify the functio
...
-|[`pre-keda-manager-operator-tests`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L127)|true|Executes basic create/update/delete functional tests of Keda Manager's reconciliation logic.|
-|[`pre-main-keda-manager-verify`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L175)|true|Installs Keda Manager, not using Lifecycle Manager, and applies the sample Keda CR in a k3d cluster. Executes smoke integration test of KEDA.  |
-|[`pre-keda-manager-operator-lint`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L61)|false|Is responsible for linting, static code analysis.|
+|[`Keda verify / lint`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L19)|true|Is responsible for the Operator linting and static code analysis.|
```
Comment (IwonaLanger) [inline docs/contributor/04-10-ci-cd.md:9]:
```suggestion
|[`Keda verify / lint`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L19)|true|Responsible for the Operator linting and static code analysis.|
```

File: docs/contributor/04-10-ci-cd.md [hunk 3]
```diff
@@ -6,19 +6,21 @@ The following CI jobs are part of the development cycle. They verify the functio
...
+|[`PR Markdown Link Check / markdown-link-check (pull_request)`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/pr-markdown-link-check.yaml)|true|Checks if there are no broken links in the pull request `.md` files.|
+|[`Keda verify / unit-test`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L29)|true|Runs basic unit tests of Operator's logic.|
+|[`Keda verify / integration-test`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L38)|true|Runs the basic functionality integration test suite for the Operator in a k3d cluster.|
```
Comment (IwonaLanger) [inline docs/contributor/04-10-ci-cd.md:12]: What `Operator` do you mean? Is it `Operator` or `the Operator`?

File: docs/contributor/04-10-ci-cd.md [hunk 4]
```diff
@@ -6,19 +6,21 @@ The following CI jobs are part of the development cycle. They verify the functio
...
+|[`pre-keda-manager-operator-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/keda-manager.yaml#L43)|true|Builds Keda operator's image and pushes it to the `dev` registry.|
+|[`pre-keda-manager-presubmit-scanner`](https://github.tools.sap/kyma/security-scanners#readme)|true|Checks if the repository doesn't contain any vulnerabilities.|
```
Comment (IwonaLanger) [inline docs/contributor/04-10-ci-cd.md:14]:
```suggestion
|[`pre-keda-manager-presubmit-scanner`](https://github.tools.sap/kyma/security-scanners#readme)|true|Checks if the repository contains any vulnerabilities.|
```

File: docs/contributor/04-10-ci-cd.md [hunk 5]
```diff
@@ -6,19 +6,21 @@ The following CI jobs are part of the development cycle. They verify the functio
...
-|[`pre-keda-manager-operator-tests`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L127)|true|Executes basic create/update/delete functional tests of Keda Manager's reconciliation logic.|
-|[`pre-main-keda-manager-verify`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L175)|true|Installs Keda Manager, not using Lifecycle Manager, and applies the sample Keda CR in a k3d cluster. Executes smoke integration test of KEDA.  |
-|[`pre-keda-manager-operator-lint`](https://github.com/kyma-project/test-infra/blob/main/templates/data/generic_module_data.yaml#L61)|false|Is responsible for linting, static code analysis.|
+|[`Keda verify / lint`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L19)|true|Is responsible for the Operator linting and static code analysis.|
+|[`PR Markdown Link Check / markdown-link-check (pull_request)`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/pr-markdown-link-check.yaml)|true|Checks if there are no broken links in the pull request `.md` files.|
```
Comment (IwonaLanger) [inline docs/contributor/04-10-ci-cd.md:10]:
```suggestion
|[`PR Markdown Link Check / markdown-link-check (pull_request)`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/pr-markdown-link-check.yaml)|true|Checks if there are any broken links in `.md` files included in the pull request .|
```

File: docs/contributor/04-10-ci-cd.md [hunk 6]
```diff
@@ -6,19 +6,21 @@ The following CI jobs are part of the development cycle. They verify the functio
...
+|[`Keda verify / unit-test`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L29)|true|Runs basic unit tests of Operator's logic.|
+|[`Keda verify / integration-test`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L38)|true|Runs the basic functionality integration test suite for the Operator in a k3d cluster.|
+|[`pre-keda-manager-operator-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/keda-manager.yaml#L43)|true|Builds Keda operator's image and pushes it to the `dev` registry.|
+|[`pre-keda-manager-presubmit-scanner`](https://github.tools.sap/kyma/security-scanners#readme)|true|Checks if the repository doesn't contain any vulnerabilities.|
 
 ## Pipelines Running on Main Branch
```
Comment (IwonaLanger) [inline docs/contributor/04-10-ci-cd.md:16]:
```suggestion
## Pipelines Running on `main` Branch 
```

File: docs/contributor/04-10-ci-cd.md (hunk 3 above)
Comment (grego952) [reply docs/contributor/04-10-ci-cd.md:12]: It's Keda Operator, so I think it's `the`

File: docs/contributor/04-10-ci-cd.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (grego952) [reply docs/contributor/04-10-ci-cd.md:14]: Should not, I'll fix it :)

File: docs/contributor/04-10-ci-cd.md (hunk 3 above)
Comment (grego952) [reply docs/contributor/04-10-ci-cd.md]: Actually, should be without `the` :D I've fixed that :)

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**